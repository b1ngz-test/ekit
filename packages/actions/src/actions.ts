import { action } from 'typesafe-actions';

// @fix: 解决由于 reselect 一份 cache 造成的 actions 更新的问题
export const createAction = action;
export * from 'redux-actions';
export { handleActions, Action } from 'redux-actions'; // @fix ts2742 in createModel.ts by export * from 'redux-actions'
export interface ActionWithPayload<P> {
  type: string;
  payload: P;
}
