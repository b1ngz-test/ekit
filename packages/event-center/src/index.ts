/**
 * @file: 事件中心
 * @author: yangqianjun
 */

import EventEmitter from 'eventemitter3';

export default new EventEmitter<string>();
