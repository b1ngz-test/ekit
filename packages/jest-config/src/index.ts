import path from 'path';
import { InitialOptions } from '@jest/types/build/Config';

export const config: Partial<InitialOptions> = (module.exports = {
  collectCoverageFrom: ['src/**/*.{js,jsx,ts,tsx}'],
  setupFiles: [],
  testMatch: [
    '<rootDir>/src/**/__tests__/**/*.(j|t)s?(x)',
    '<rootDir>/tests/**/?(*.)(spec|test).(j|t)s?(x)',
    '<rootDir>/__tests__/**/?(*.)(spec|test).(j|t)s?(x)'
  ],
  testEnvironment: 'node',
  testURL: 'http://localhost:4444',
  transform: {
    '^.+\\.(js|jsx|mjs)$': 'babel-jest',
    '^.+\\.tsx?$': 'ts-jest'
  },
  transformIgnorePatterns: ['[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs|ts|tsx)$'],
  moduleNameMapper: {
    '^src(.*)$': '<rootDir>/src/$1',
    'src(.*)$': '<rootDir>/src/$1'
  },
  moduleFileExtensions: [
    'h5.ts',
    'web.ts',
    'ts',
    'h5.tsx',
    'web.tsx',
    'tsx',
    'web.js',
    'js',
    'web.jsx',
    'jsx',
    'json',
    'node',
    'mjs'
  ],
  globals: {
    'ts-jest': {
      tsConfig: path.join(process.cwd(), 'tsconfig.test.json')
    }
  }
});
