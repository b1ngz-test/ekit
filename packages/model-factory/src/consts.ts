/**
 * @file: model 常量
 * @author: yangqianjun
 * @Date: 2020-02-06 14:42:35
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-10 10:06:56
 */

import { EffectOptions } from './types';

/** 默认副作用配置 */
export const defaultEffectOptions: EffectOptions = {
  type: 'takeEvery',
  ms: 100
};

/** 如引入 immer，枚举标记 redux 或者 react hooks */
export const enum ModernType {
  /** redux */
  ReduxModern = 0,
  /** useModel */
  HookModern = 1
}

/**
 * 执行 effect 广播的事件
 * 从 @ekit/async 拆过来，以逐步解除 model & async 的循环引用
 */
export const ASYNC_EFFECT_EVENT_NAME = 'ASYNC_EFFECT_EVENT_NAME';

/** Redux Type 副作用开始 */
export const EFFECTS_START = 'EFFECTS_START';

/** Redux Type 副作用结束 */
export const EFFECTS_END = 'EFFECTS_END';

/** Redux Type 副作用错误 */
export const EFFECTS_ERROR = 'EFFECTS_ERROR';

/** Redux Type 副作用成功 */
export const EFFECTS_SUCCESS = 'EFFECTS_SUCCESS';

/**  副作用消息EventCenter事件 */
export const EFFECTS_MSG = 'EFFECTS_MSG';

export const fakeEffectRes = { code: 0 };

/** 副作用返回的promise的resolve */
export const EFFECTS_PROMISE_RESOLVE = '__EFFECTS_PROMISE_RESOLVE__';

/** 副作用返回的promise的resolve */
export const EFFECTS_PROMISE_REJECT = '__EFFECTS_PROMISE_REJECT__';

/** 是否 tkit effet */
export const TKIT_EFFECT = '__TKIT_EFFECT__';

/** 是否 Tkit Non blocking Effect */
export const TKIT_NB_EFFECT = '__TKIT_NB_EFFECT__';

/** 标记是子 effect，不自动捕获错误以及发起 EFFECT START */
export const TKIT_SUB_EFFECT = '__TKIT_SUB_EFFECT__';

/** 注入 tDispath */
export const TKIT_DISPATCH = '__TKIT_DISPATCH__';

/** 注入 get State */
export const TKIT_GET_STATE = '__TKIT_GET_STATE__';

/** 为 action 标记子 effect 标记  */
export const markSubEffect = <A>(action: A) => {
  action[TKIT_SUB_EFFECT] = true;
};

export const noop = () => void 0;
export const fakeUtilsThrowErrorIfInvoked = () => {
  throw 'Fake Utils mustn\'t be Invoked';
};
export const printError = (message: string) => {
  // TODO: UC Browser for Android 不支持 console.error
  // eslint-disable-next-line no-console
  if (typeof console.error === 'function') {
    // eslint-disable-next-line no-console
    console.error(message);
  }
};
