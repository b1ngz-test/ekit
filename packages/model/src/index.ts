/**
 * @file: description
 * @author: yangqianjun
 * @Date: 2019-07-20 15:12:00
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-07 16:35:02
 */

export * from '@ekit/model-factory';
export { M, MM, useModel } from '@ekit/use-model';
export { CM } from '@ekit/redux-model';
export { default, promiseMiddleware } from '@ekit/redux-model';
