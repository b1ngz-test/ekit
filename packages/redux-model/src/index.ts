/**
 * @file: description
 * @author: yangqianjun
 * @Date: 2019-07-20 15:12:00
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-07 17:18:36
 */

export * from 'redux-saga/effects';
export * from '@ekit/model-factory';
export * from './cm';
export * from './createModel';
export * from './promiseMiddleware';
export { default } from './createModel';
