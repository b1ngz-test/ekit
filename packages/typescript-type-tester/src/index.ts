import { parser } from './typescript-type-tester';
export { parser, parser as default };
