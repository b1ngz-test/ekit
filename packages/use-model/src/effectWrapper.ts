import {
  EFFECTS_START,
  EFFECTS_END,
  FakeEffectFactory,
  EFFECTS_ERROR,
  IEffectWrapper,
  Tction,
  EFFECTS_PROMISE_RESOLVE,
  EFFECTS_PROMISE_REJECT
} from '@ekit/model-factory';

/** for Hooks model */
export const effectWrapper: IEffectWrapper = (effect, effects, effectName, options) => {
  const { tPut: put } = effects;
  // 默认不展示loading
  const loading = (options && options.loading) || false;
  const silent = options && options.silent;
  // 局部Model不再使用gerator
  return async (action: Tction<any>) => {
    await put({ type: EFFECTS_START, payload: { effectName } } as any);
    const { getResolver, getRes, getEffect } = FakeEffectFactory({
      effectName,
      silent,
      loading
    });
    getEffect()();
    const res = getRes();
    try {
      if (typeof effect === 'function') {
        res.message = (await effect(effects, action)) || false;
        if (action[EFFECTS_PROMISE_RESOLVE]) {
          action[EFFECTS_PROMISE_RESOLVE](res.message);
        }
      }
    } catch (e) {
      res.code = 10100;
      res.message = e && e['message'] ? e['message'] : e;
      await put({ type: EFFECTS_ERROR, payload: { effectName } } as any);
      if (action[EFFECTS_PROMISE_REJECT]) {
        action[EFFECTS_PROMISE_REJECT](res.message);
      }
    }
    await put({ type: EFFECTS_END, payload: { effectName } } as any);
    // 隐藏 loading
    getResolver()(res);
  };
};
