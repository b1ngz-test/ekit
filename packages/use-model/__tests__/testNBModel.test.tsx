/* eslint-disable @typescript-eslint/camelcase */
import React, { useEffect } from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { useModel, MM, Tction } from 'src/index';

const originalLogError = console.error;

const genAssets = () => {
  const logsArr: string[] = [];
  const logger = jest.fn((log: string) => {
    logsArr.push(log);
  });
  console.error = jest.fn((...args: any[]) => {
    logger(args[0]);
    return originalLogError.apply(console, args);
  });
  const UserMMNBModel = MM({
    namespace: 'test',
    state: {
      username: ''
    },
    reducers: {
      doRename: (state, action: Tction<{ username: string }>) => {
        state.username = action.payload.username;
      }
    },
    effects: {
      async doNameOK(utils, action: Tction<string>) {
        await new Promise(rs => setTimeout(rs, 50));
        logger(`doNameOK: ${action.payload}`);
        return `OK doNameOK: ${action.payload}`;
      },
      async doNameERROR(utils, action: Tction<string>) {
        await new Promise(rs => setTimeout(rs, 50));
        logger(`doNameERROR: ${action.payload}`);
        throw `ERROR doNameERROR: ${action.payload}`;
      },
      async doNonBlockingOK_NonBlockingERROR(utils, action: Tction<string>) {
        let res: any = await utils.tNBPut(UserMMNBModel.actions.doNameOK, `${action.payload} -1`);
        logger(`doNameOK: ${action.payload} 1 ${JSON.stringify(res)}`);
        res = await utils.tNBPut(UserMMNBModel.actions.doNameERROR, `${action.payload} -1`);
        logger(`doNameERROR: ${action.payload} 2 ${JSON.stringify(res)}`);
        return `OK: doNonBlockingOK_NonBlockingERROR`;
      },
      async doNonBlockingOK_NonBlockingERROR_ERROR(utils, action: Tction<string>) {
        let res: any = await utils.tNBPut(UserMMNBModel.actions.doNameOK, `${action.payload} -1`);
        logger(`doNameOK: ${action.payload} ${JSON.stringify(res)} 1`);
        res = await utils.tNBPut(UserMMNBModel.actions.doNameERROR, `${action.payload} -1`);
        logger(`doNameERROR: ${action.payload} ${JSON.stringify(res)} 2`);
        throw `ERROR: doNonBlockingOK_NonBlockingERROR_ERROR`;
      },
      async doNonBlockingOK_BlockingERROR(utils, action: Tction<string>) {
        let res: any = await utils.tNBPut(UserMMNBModel.actions.doNameOK, `${action.payload} 2`);
        logger(`doNameOK: ${action.payload} ${JSON.stringify(res)} 1`);
        res = await utils.tPut(UserMMNBModel.actions.doNameERROR, `${action.payload} 3`);
        logger(`doNameERROR: ${action.payload} ${JSON.stringify(res)} -1`);
      },
      async doBlockingOK_BlockingERROR(utils, action: Tction<string>) {
        let res: any = await utils.tPut(UserMMNBModel.actions.doNameOK, `${action.payload} 1`);
        logger(`doNameOK: ${action.payload} ${JSON.stringify(res)} 2`);
        res = await utils.tPut(UserMMNBModel.actions.doNameERROR, `${action.payload} 3`);
        logger(`doNameERROR: ${action.payload} ${JSON.stringify(res)} -1`);
      },
      async doBlockingOK_NonBlockingERROR(utils, action: Tction<string>) {
        let res: any = await utils.tPut(UserMMNBModel.actions.doNameOK, `${action.payload} 1`);
        logger(`doNameERROR: ${action.payload} ${JSON.stringify(res)} 2`);
        res = await utils.tNBPut(UserMMNBModel.actions.doNameERROR, `${action.payload} 4`);
        logger(`doNameERROR: ${action.payload} ${JSON.stringify(res)} 3`);
      }
    }
  });
  return {
    logger,
    logsArr,
    model: UserMMNBModel
  };
};

let assets: ReturnType<typeof genAssets>;

describe('utils/*Model work ok', () => {
  let container: HTMLDivElement;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    assets = genAssets();
    window.__TKIT_USE_MODEL_LOGGER__ = jest.fn((...args) => {
      assets.logger(`devtool watched action: ${JSON.stringify(args[3])}`);
    });
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null as any;
  });

  // checked
  1 &&
    it('useModel doNonBlockingOK_NonBlockingERROR work ok', async () => {
      let prom: any;
      const Cp = () => {
        const [store, localActions] = useModel(assets.model);
        return (
          <button
            onClick={() => {
              prom = localActions
                .doNonBlockingOK_NonBlockingERROR('doNonBlockingOK_NonBlockingERROR')
                .then(
                  res => {
                    //
                    assets.logger(`Promise OK: ${res}`);
                  },
                  e => {
                    //
                    assets.logger(`Promise ERROR: ${(e && e['message']) || e}`);
                  }
                );
            }}
          >
            {store.username}
          </button>
        );
      };
      act(() => {
        ReactDOM.render(<Cp />, container);
      });

      const button = container.querySelector('button') as HTMLButtonElement;
      act(() => {
        button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      // @cc: 预估的 10ms delay，直到 effect 被调用
      await new Promise(rs => setTimeout(rs, 10));
      await prom;
      await new Promise(rs => setTimeout(rs, 100));
      expect(assets.logsArr).toMatchSnapshot('doNonBlockingOK_NonBlockingERROR');
      expect(assets.logger).toBeCalledTimes(16);
      expect(assets.logsArr).toHaveLength(16);
    });

  // checked
  1 &&
    it('useModel doNonBlockingOK_NonBlockingERROR_ERROR work ok', async () => {
      let prom: any;
      const Cp = () => {
        const [store, localActions] = useModel(assets.model);
        return (
          <button
            onClick={() => {
              prom = localActions
                .doNonBlockingOK_NonBlockingERROR_ERROR('doNonBlockingOK_NonBlockingERROR_ERROR')
                .then(
                  res => {
                    //
                    assets.logger(`Promise OK: ${res}`);
                  },
                  e => {
                    //
                    assets.logger(`Promise ERROR: ${(e && e['message']) || e}`);
                  }
                );
            }}
          >
            {store.username}
          </button>
        );
      };
      act(() => {
        ReactDOM.render(<Cp />, container);
      });

      const button = container.querySelector('button') as HTMLButtonElement;
      act(() => {
        button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      // @cc: 预估的 10ms delay，直到 effect 被调用
      await new Promise(rs => setTimeout(rs, 10));
      await prom;
      await new Promise(rs => setTimeout(rs, 100));
      expect(assets.logsArr).toMatchSnapshot('doNonBlockingOK_NonBlockingERROR_ERROR');
      expect(assets.logsArr).toHaveLength(18);
      expect(assets.logger).toBeCalledTimes(18);
    });

  // checked
  1 &&
    it('useModel doNonBlockingOK_BlockingERROR work ok', async () => {
      let prom: any;
      const Cp = () => {
        const [store, localActions] = useModel(assets.model);
        return (
          <button
            onClick={() => {
              prom = localActions
                .doNonBlockingOK_BlockingERROR('doNonBlockingOK_BlockingERROR')
                .then(
                  res => {
                    //
                    assets.logger(`Promise OK: ${res}`);
                  },
                  e => {
                    //
                    assets.logger(`Promise ERROR: ${(e && e['message']) || e}`);
                  }
                )
                .catch();
            }}
          >
            {store.username}
          </button>
        );
      };
      act(() => {
        ReactDOM.render(<Cp />, container);
      });

      const button = container.querySelector('button') as HTMLButtonElement;
      act(() => {
        button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      // @cc: 预估的 10ms delay，直到 effect 被调用
      await new Promise(rs => setTimeout(rs, 10));
      await prom;
      await new Promise(rs => setTimeout(rs, 100));
      expect(assets.logsArr).toMatchSnapshot('doNonBlockingOK_BlockingERROR');
      expect(assets.logsArr).toHaveLength(14);
      expect(assets.logger).toBeCalledTimes(14);
    });

  // checked
  1 &&
    it('useModel doBlockingOK_BlockingERROR work ok', async () => {
      let prom: any;
      const Cp = () => {
        const [store, localActions] = useModel(assets.model);
        return (
          <button
            onClick={() => {
              prom = localActions.doBlockingOK_BlockingERROR('doBlockingOK_BlockingERROR').then(
                res => {
                  //
                  assets.logger(`Promise OK: ${res}`);
                },
                e => {
                  //
                  assets.logger(`Promise ERROR: ${(e && e['message']) || e}`);
                }
              );
            }}
          >
            {store.username}
          </button>
        );
      };
      act(() => {
        ReactDOM.render(<Cp />, container);
      });

      const button = container.querySelector('button') as HTMLButtonElement;
      act(() => {
        button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      // @cc: 预估的 10ms delay，直到 effect 被调用
      await new Promise(rs => setTimeout(rs, 10));
      await prom;
      await new Promise(rs => setTimeout(rs, 100));
      expect(assets.logsArr).toMatchSnapshot('doBlockingOK_BlockingERROR');
      expect(assets.logsArr).toHaveLength(12);
      expect(assets.logger).toBeCalledTimes(12);
    });

  // checked
  1 &&
    it('useModel doBlockingOK_NonBlockingERROR work ok', async () => {
      let prom: any;
      const Cp = () => {
        const [store, localActions] = useModel(assets.model);
        return (
          <button
            onClick={() => {
              prom = localActions
                .doBlockingOK_NonBlockingERROR('doBlockingOK_NonBlockingERROR')
                .then(
                  res => {
                    //
                    assets.logger(`Promise OK: ${res}`);
                  },
                  e => {
                    //
                    assets.logger(`Promise ERROR: ${(e && e['message']) || e}`);
                  }
                );
            }}
          >
            {store.username}
          </button>
        );
      };
      act(() => {
        ReactDOM.render(<Cp />, container);
      });

      const button = container.querySelector('button') as HTMLButtonElement;
      act(() => {
        button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });
      // @cc: 预估的 10ms delay，直到 effect 被调用
      await new Promise(rs => setTimeout(rs, 10));
      await prom;
      await new Promise(rs => setTimeout(rs, 100));
      expect(assets.logsArr).toMatchSnapshot('doBlockingOK_NonBlockingERROR');
      expect(assets.logsArr).toHaveLength(14);
      expect(assets.logger).toBeCalledTimes(14);
    });
});
