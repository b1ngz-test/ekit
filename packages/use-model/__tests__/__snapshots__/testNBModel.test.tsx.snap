// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/*Model work ok useModel doBlockingOK_BlockingERROR work ok: doBlockingOK_BlockingERROR 1`] = `
Array [
  "devtool watched action: {\\"type\\":\\"test/doBlockingOK_BlockingERROR\\",\\"payload\\":\\"doBlockingOK_BlockingERROR\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doBlockingOK_BlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doBlockingOK_BlockingERROR 1\\",\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doBlockingOK_BlockingERROR 1",
  "doNameOK: doBlockingOK_BlockingERROR \\"OK doNameOK: doBlockingOK_BlockingERROR 1\\" 2",
  "devtool watched action: {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doBlockingOK_BlockingERROR 3\\",\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameERROR: doBlockingOK_BlockingERROR 3",
  "HOOKS_EFFECTS_ERROR: test/doNameERROR ERROR doNameERROR: doBlockingOK_BlockingERROR 3",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doBlockingOK_BlockingERROR\\",\\"error\\":\\"ERROR doNameERROR: doBlockingOK_BlockingERROR 3\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doBlockingOK_BlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doBlockingOK_BlockingERROR ERROR doNameERROR: doBlockingOK_BlockingERROR 3",
  "Promise ERROR: ERROR doNameERROR: doBlockingOK_BlockingERROR 3",
]
`;

exports[`utils/*Model work ok useModel doBlockingOK_NonBlockingERROR work ok: doBlockingOK_NonBlockingERROR 1`] = `
Array [
  "devtool watched action: {\\"type\\":\\"test/doBlockingOK_NonBlockingERROR\\",\\"payload\\":\\"doBlockingOK_NonBlockingERROR\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doBlockingOK_NonBlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doBlockingOK_NonBlockingERROR 1\\",\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doBlockingOK_NonBlockingERROR 1",
  "doNameERROR: doBlockingOK_NonBlockingERROR \\"OK doNameOK: doBlockingOK_NonBlockingERROR 1\\" 2",
  "devtool watched action: {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doBlockingOK_NonBlockingERROR 4\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameERROR: doBlockingOK_NonBlockingERROR {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doBlockingOK_NonBlockingERROR 4\\"} 3",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doBlockingOK_NonBlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "Promise OK: false",
  "doNameERROR: doBlockingOK_NonBlockingERROR 4",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\",\\"error\\":\\"ERROR doNameERROR: doBlockingOK_NonBlockingERROR 4\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNameERROR ERROR doNameERROR: doBlockingOK_NonBlockingERROR 4",
]
`;

exports[`utils/*Model work ok useModel doNonBlockingOK_BlockingERROR work ok: doNonBlockingOK_BlockingERROR 1`] = `
Array [
  "devtool watched action: {\\"type\\":\\"test/doNonBlockingOK_BlockingERROR\\",\\"payload\\":\\"doNonBlockingOK_BlockingERROR\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_BlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_BlockingERROR 2\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doNonBlockingOK_BlockingERROR {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_BlockingERROR 2\\"} 1",
  "devtool watched action: {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doNonBlockingOK_BlockingERROR 3\\",\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doNonBlockingOK_BlockingERROR 2",
  "doNameERROR: doNonBlockingOK_BlockingERROR 3",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNameERROR ERROR doNameERROR: doNonBlockingOK_BlockingERROR 3",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_BlockingERROR\\",\\"error\\":\\"ERROR doNameERROR: doNonBlockingOK_BlockingERROR 3\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_BlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNonBlockingOK_BlockingERROR ERROR doNameERROR: doNonBlockingOK_BlockingERROR 3",
  "Promise ERROR: ERROR doNameERROR: doNonBlockingOK_BlockingERROR 3",
]
`;

exports[`utils/*Model work ok useModel doNonBlockingOK_NonBlockingERROR work ok: doNonBlockingOK_NonBlockingERROR 1`] = `
Array [
  "devtool watched action: {\\"type\\":\\"test/doNonBlockingOK_NonBlockingERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_NonBlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doNonBlockingOK_NonBlockingERROR 1 {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameERROR: doNonBlockingOK_NonBlockingERROR 2 {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_NonBlockingERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "Promise OK: OK: doNonBlockingOK_NonBlockingERROR",
  "doNameOK: doNonBlockingOK_NonBlockingERROR -1",
  "doNameERROR: doNonBlockingOK_NonBlockingERROR -1",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\",\\"error\\":\\"ERROR doNameERROR: doNonBlockingOK_NonBlockingERROR -1\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNameERROR ERROR doNameERROR: doNonBlockingOK_NonBlockingERROR -1",
]
`;

exports[`utils/*Model work ok useModel doNonBlockingOK_NonBlockingERROR_ERROR work ok: doNonBlockingOK_NonBlockingERROR_ERROR 1`] = `
Array [
  "devtool watched action: {\\"type\\":\\"test/doNonBlockingOK_NonBlockingERROR_ERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR_ERROR\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_NonBlockingERROR_ERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR_ERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameOK: doNonBlockingOK_NonBlockingERROR_ERROR {\\"type\\":\\"test/doNameOK\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR_ERROR -1\\"} 1",
  "devtool watched action: {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR_ERROR -1\\"}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_START\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "doNameERROR: doNonBlockingOK_NonBlockingERROR_ERROR {\\"type\\":\\"test/doNameERROR\\",\\"payload\\":\\"doNonBlockingOK_NonBlockingERROR_ERROR -1\\"} 2",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_NonBlockingERROR_ERROR\\",\\"error\\":\\"ERROR: doNonBlockingOK_NonBlockingERROR_ERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNonBlockingOK_NonBlockingERROR_ERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNonBlockingOK_NonBlockingERROR_ERROR ERROR: doNonBlockingOK_NonBlockingERROR_ERROR",
  "Promise ERROR: ERROR: doNonBlockingOK_NonBlockingERROR_ERROR",
  "doNameOK: doNonBlockingOK_NonBlockingERROR_ERROR -1",
  "doNameERROR: doNonBlockingOK_NonBlockingERROR_ERROR -1",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameOK\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_ERROR\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\",\\"error\\":\\"ERROR doNameERROR: doNonBlockingOK_NonBlockingERROR_ERROR -1\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "devtool watched action: {\\"type\\":\\"EFFECTS_END\\",\\"payload\\":{\\"effectName\\":\\"test/doNameERROR\\"},\\"__TKIT_SUB_EFFECT__\\":true}",
  "HOOKS_EFFECTS_ERROR: test/doNameERROR ERROR doNameERROR: doNonBlockingOK_NonBlockingERROR_ERROR -1",
]
`;
